src/plugins/intel_cpu/src/nodes/mkldnn_matmul_node.cpp

```diff
@@ -307,6 +308,80 @@ void MKLDNNMatMulNode::getSupportedDescriptors() {
     createDescriptor({inDataDesc[0], inDataDesc[1]}, {outDataDesc});
 }
 
+std::pair<Shape, Shape> MKLDNNMatMulNode::makeDummyInputShapes(const Shape& in0, const Shape& in1) const {
```

Comment:
I'm just wondering what if we use some canonical axis order so that each dim in one shape corresponds to the same order dim in another tensor, so we can use one pass algorithm without this cumbersome constructions, something like that {b1, b2, b3, m, k} * {b4, b5, b6, k, n} - transform to the canonical form -> {b1, b2, b3, k, m} *  {b4, b5, b6, k, n} - one pass dimensions alignment-> backward transform of the aligned dimensions -> PROFIT

Reply:
If I understand your idea correctly the only difference between current algorithm and algorithm which you suggested will that we will be iterate up to `inDims0.size() - 1` not `inDims0.size() - 2` here: https://github.com/openvinotoolkit/openvino/pull/9858/files#diff-4bb2f96c6c464c7633a17f418a81f01a89152ec5f19eb6b4e0c69ca1b961421bR376

if you mean that `fillDummy` is cumbersome construction, its cumbersome is not depend on dims position

if I missed something correct me

Reply:
Yes, the idea is to preprocess input shapes in such a way that we can call `fillDummy` in the loop for all axis (btw in that case we can inline the lambda), without addition special k,m,n processing. `fillDummy ` algo is complicated because the `Shape` class does not have ranges abstraction, so we can not simplify it.  What do you think?

Reply:
we still need to special call for m,n because they doesn't depend on each other and we can't call `fillDummy` for them

about transformation to `{b1, b2, b3, k, m} * {b4, b5, b6, k, n}`
in this case we will call `swapTranspDims` to align  k, m, n positions and after that swap k - m
or add `if - else` to take into account `transposeIn` values

I can apply first solution but from my point of view it's surplus because because benefit which we get is  iteratinf up to `inDims0.size() - 1` not `inDims0.size() - 2` but we get more complex preprocessing. We need to perform these (`in this case we will call `swapTranspDims` to align  k, m, n positions and after that swap k - m`) actions in reverse order

Reply:
Ok, let's leave it as it is.